app/server/utils/blockchain_tasks.py

```diff
@@ -1,280 +1,446 @@
+from typing import Optional, List, Dict
+from functools import partial
 from flask import current_app
-from server import celery_app
 from eth_keys import keys
 from eth_utils import keccak
-import os, random
-
-eth_worker_name = 'eth_manager'
-celery_tasks_name = 'celery_tasks'
-eth_endpoint = lambda endpoint: f'{eth_worker_name}.{celery_tasks_name}.{endpoint}'
+import os
+import random
+from time import sleep
 
-def _execute_synchronous_celery(signature):
-    async_result = signature.delay()
-
-    try:
-        response = async_result.get(timeout=20, propagate=True, interval=0.3)
-    except Exception as e:
-        raise e
-    finally:
-        async_result.forget()
+from server import celery_app
+from server.utils.exchange import (
+    bonding_curve_tokens_to_reserve,
+    bonding_curve_reserve_to_tokens,
+    bonding_curve_token1_to_token2
+)
 
-    return response
 
-def execute_synchronous_transaction_task(signature):
-    if current_app.config['IS_TEST']:
-        # TODO: We need a better way of stubbing responses from the blockchain worker during tests
-        return random.randint(0, 1000000000)
+class BlockchainTasker(object):
```

Comment:
shifted all of these into a class to make it easier to mock during testing with monkeypatching

Reply:
ah bt = blockchain tasker :)